torch/distributed/distributed_c10d.py

```diff
@@ -172,6 +172,15 @@ class GroupMember(object):
 # Process group count for default naming
 _group_count = 0
 
+STORE_BASED_BARRIER_PREFIX = "store_based_barrier_key"
+
+def _store_based_barrier(rank, store):
+    global _group_count
+    store_key = "{}:{}".format(STORE_BASED_BARRIER_PREFIX, _group_count)
+    if rank == 0:
+        store.set(store_key, "1")
```

Comment:
This probably won't work since rank 0 doesn't wait for all other ranks, will fix this.